psgo.go

```diff
@@ -334,19 +334,68 @@ func JoinNamespaceAndProcessInfo(pid string, descriptors []string) ([][]string,
 	return data, dataErr
 }
 
+// JoinNamespaceAndProcessInfoByPids has similar semantics to
+// JoinNamespaceAndProcessInfo and avoids duplicate entries by joining a giving
+// PID namepsace only once.
+func JoinNamespaceAndProcessInfoByPids(pids []string, descriptors []string) ([][]string, error) {
+	// Extracting data from processes that share the same PID namespace
+	// would yield duplicate results.  Avoid that by extracting data only
+	// from the first process in `pids` from a given PID namespace.
+	// `nsMap` is used for quick lookups if a given PID namespace is
+	// already covered, `pidList` is used to preserve the order which is
+	// not guaranteed by nondeterministic maps in golang.
+	nsMap := make(map[string]bool)
+	pidList := []string{}
+	for _, pid := range pids {
+		ns, err := proc.ParsePIDNamespace(pid)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error extracing PID namespace")
```

Comment:
Should this be a warning?  Seems like this is going to be racy, and returning errors without all of the info might be a pain in the butt.

Reply:
Thanks for catching this! We do an `os.IsNotExist(err) { continue }` at all other places. I changed it to do the same. But logging may be nice to have at some point.